pkg/virtualKubelet/liqoNodeProvider/cordon.go

```diff
@@ -0,0 +1,49 @@
+package liqonodeprovider
+
+import (
+	"context"
+	"encoding/json"
+
+	"gomodules.xyz/jsonpatch/v2"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/klog/v2"
+)
+
+// cordonNode cordons the controlled node setting it in the unschedulable state.
+func (p *LiqoNodeProvider) cordonNode(ctx context.Context) error {
```

Comment:
You are already using a very similar logic to update the labels of the node. I would suggest creating a generic function which given the original and patched nodes performs the actual patch, to avoid duplicating the marshal, createpatch, patch logic.